frontend/pages/approve.tsx

```diff
@@ -13,77 +13,132 @@ import {
   STAKING_ADDRESS,
 } from '@pythnetwork/staking'
 import { useEffect, useState } from 'react'
-import { utils, Wallet } from '@coral-xyz/anchor'
-import toast from 'react-hot-toast'
-import { capitalizeFirstLetter } from '../utils/capitalizeFirstLetter'
+import { Wallet } from '@coral-xyz/anchor'
 import { PublicKey } from '@solana/web3.js'
 import { useRouter } from 'next/router'
+import toast from 'react-hot-toast'
+import { capitalizeFirstLetter } from 'utils/capitalizeFirstLetter'
 
 const ApproveSplit: NextPage = () => {
   const { connection } = useConnection()
   const anchorWallet = useAnchorWallet()
-  const { publicKey, connected } = useWallet()
+  const [amount, setAmount] = useState<PythBalance>()
```

Comment:
Can combine amount and recipient in an object. Since we are setting them always together. 

Having an object result in one state update. Else, it will be multiple state updates. Imo is a good practice to have as less state update as possible.